configure.ac

```diff
@@ -264,12 +168,121 @@ Either DirectWrite (on Windows), CoreText (on OSX), or Fontconfig \
+            BITS=64
+            BITTYPE=x32
+            ASFLAGS="$ASFLAGS -DARCH_X86_64=1 -DPIC" ],
+        [x86_64-*|amd64-*], [
+            AS=nasm
+            INTEL=true
+            X64=true
+            BITS=64
+            BITTYPE=64
+            ASFLAGS="$ASFLAGS -DARCH_X86_64=1 -DPIC" ],
+        )
+    AS_IF([test x$INTEL = xtrue], [
+        AC_CHECK_PROG([nasm_check], [$AS], [yes])
+        AS_IF([test x$nasm_check != xyes], [
+            AC_MSG_WARN(nasm was not found; ASM functions are disabled.)
+            AC_MSG_WARN(Install nasm for a significantly faster libass build.)
+            enable_asm=no
+        ], [
+            AS_CASE([$host],
+                [*darwin*], [
+                    ASFLAGS="$ASFLAGS -f macho$BITTYPE -DPREFIX -DHAVE_ALIGNED_STACK=1" ],
+                [*linux*|*dragonfly*|*bsd*|*solaris*|*haiku*], [
+                    ASFLAGS="$ASFLAGS -f elf$BITTYPE -DHAVE_ALIGNED_STACK=1" ],
+                [*cygwin*|*mingw*], [
+                    ASFLAGS="$ASFLAGS -f win$BITTYPE"
+                    AS_IF([test x$BITS = x64], [
+                        ASFLAGS="$ASFLAGS -DHAVE_ALIGNED_STACK=1"
+                    ], [
+                        ASFLAGS="$ASFLAGS -DHAVE_ALIGNED_STACK=0 -DPREFIX"
+                    ])
+                ],
+                [ #default
+                    AC_MSG_ERROR(m4_text_wrap(m4_normalize([
+                            Please contact libass upstream to figure out if ASM
+                            support for your platform can be added.
+                            In the meantime you will need to use --disable-asm.]),
+                        [                  ],
+                        [could not identify NASM format for $host !],
+                        [78])
+                    )
+                ]
+            )
+            ASFLAGS="$ASFLAGS -DHAVE_CPUNOP=0 -Dprivate_prefix=ass"
+            AC_MSG_CHECKING([if $AS supports vpmovzxwd])
+            echo "vpmovzxwd ymm0, xmm0" > conftest.asm
+            AS_IF([$AS conftest.asm $ASFLAGS -o conftest.o >conftest.log 2>&1], [
+                AC_MSG_RESULT([yes])
+            ], [
+                AC_MSG_RESULT([no])
+                VER=`($AS --version || echo no assembler) 2>/dev/null | head -n 1`
+                AC_MSG_WARN([nasm is too old (found $VER); ASM functions are disabled.])
+                AC_MSG_WARN([Install nasm-2.10 or later for a significantly faster libass build.])
+                enable_asm=no
+            ])
+            rm conftest.asm conftest.o > /dev/null 2>&1
+        ])
+    ])
+])
+
+## Tell Makefiles which assembler and flags to use
+AC_SUBST([ASFLAGS], ["$ASFLAGS"])
+AC_SUBST([AS], ["$AS"])
+
+
+# Relay config results to output files
+
+## Setup conditionals for use in Makefiles
+AM_CONDITIONAL([ASM], [test x$enable_asm != xno])
+AM_CONDITIONAL([INTEL], [test x$INTEL = xtrue])
+AM_CONDITIONAL([X86], [test x$X86 = xtrue])
+AM_CONDITIONAL([X64], [test x$X64 = xtrue])
```

Comment:
Should this perhaps be closer to `AC_SUBST([PKG_LIBS_DEFAULT]` etc.?

Reply:
Now it is, thanks.